[2.8.2] CMake build scripts do not detect `/dev/urandom` support on AIX?

We've been using this library for a long time, but recently switched to building it as a static library using the provided CMAKE build files, instead of 'vendoring' the source files (i.e. compiling them explicitly & adding the object files to _our_ static library)

I've notice that when we invoke it using the command `cmake -DCMAKE_C_COMPILER=ibm-clang_r -DCMAKE_BUILD_TYPE=Release -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_C_FLAGS=-m32 -DEXPAT_SHARED_LIBS=OFF -DEXPAT_BUILD_TOOLS=OFF -DEXPAT_BUILD_EXAMPLES=OFF -DEXPAT_BUILD_TESTS=OFF -DEXPAT_BUILD_DOCS=OFF -DCMAKE_ARCHIVE_OUTPUT_DIRECTORY=/home/simba/jk/sen-quickstart-cpp/dependencies/third-party/libexpat/expat/build/aix7_2_ibmclang17_1_debug32 ../..`, we get output that contains

```
[2026-08-06T03:25:29.569Z] -- Looking for getrandom
[2026-08-06T03:25:29.569Z] -- Looking for getrandom - not found
[2026-08-06T03:25:29.569Z] -- Performing Test HAVE_ARC4RANDOM_BUF
[2026-08-06T03:25:29.569Z] -- Performing Test HAVE_ARC4RANDOM_BUF - Failed
[2026-08-06T03:25:29.569Z] -- Performing Test HAVE_ARC4RANDOM
[2026-08-06T03:25:29.569Z] -- Performing Test HAVE_ARC4RANDOM - Failed
```

and then we we run `cmake --build /home/simba/jk/sen-quickstart-cpp/dependencies/third-party/libexpat/expat/lib/../build/aix7_2_ibmclang17_1_debug32 --target expat`, we get the error

```
[2026-08-06T03:25:29.572Z] /home/simba/jk/sen-quickstart-cpp/dependencies/third-party/libexpat/expat/lib/xmlparse.c:150:4: error: You do not have support for any sources of high quality entropy     enabled.  For end user security, that is probably not what you want.         Your options include:       * Linux >=3.17 + glibc >=2.25 (getrandom): HAVE_GETRANDOM,       * Linux >=3.17 + glibc (including <2.25) (syscall SYS_getrandom): HAVE_SYSCALL_GETRANDOM,       * BSD / macOS >=10.7 / glibc >=2.36 (arc4random_buf): HAVE_ARC4RANDOM_BUF,       * BSD / macOS (including <10.7) / glibc >=2.36 (arc4random): HAVE_ARC4RANDOM,       * BSD / macOS >=10.12 / glibc >=2.25 (getentropy): HAVE_GETENTROPY,       * Linux (including <3.17) / BSD / macOS (including <10.7) / Solaris >=8 (/dev/urandom): XML_DEV_URANDOM,       * Windows >=Vista (rand_s): _WIN32.         If you insist on not using any of these, bypass this error by defining     XML_POOR_ENTROPY and be vulnerable to hash flooding; you have been warned.         If you have reasons to patch this detection code away or need changes     to the build system, please open a bug.  Thank you!
[2026-08-06T03:25:29.573Z]   150 | #  error You do not have support for any sources of high quality entropy \
[2026-08-06T03:25:29.573Z]       |    ^
```

IMO it should default to using `/dev/random`, which I believe AIX has?

This is with the 2.8.2 release.

I'm going to try to rebuild w/ `-DEXPAT_DEV_URANDOM=ON` & report back whether that worked.

On a related note, I noticed the following warning in the same build, which IMO could be a big problem (UB?)

```
[2026-08-06T03:25:29.573Z] /home/simba/jk/sen-quickstart-cpp/dependencies/third-party/libexpat/expat/lib/xmlparse.c:8602:11: warning: format specifies type 'int' but the argument has type 'ptrdiff_t' (aka 'long') [-Wformat]
[2026-08-06T03:25:29.574Z]  8601 |           " (+" EXPAT_FMT_PTRDIFF_T("6") " bytes %s|%u, xmlparse.c:%d) %*s\"",
[2026-08-06T03:25:29.574Z]       |                 ~~~~~~~~~~~~~~~~~~~~~~~~
[2026-08-06T03:25:29.574Z]  8602 |           bytesMore, (account == XML_ACCOUNT_DIRECT) ? "DIR" : "EXP",
[2026-08-06T03:25:29.574Z]       |           ^~~~~~~~~
```

Hmm, looks like there's other problems; it got past that error, but immediately after failed with

```
21:35:32  [100%] Linking C static library libexpat.a
21:35:32  Could not load program /opt/freeware/bin/cmake:
21:35:32  	Dependent module libexpat.a(libexpat.so.1) could not be loaded.
21:35:32  	Member libexpat.so.1 is not found in archive 
21:35:32  gmake[4]: *** [CMakeFiles/Makefile2:88: CMakeFiles/expat.dir/all] Error 255
21:35:32  gmake[3]: *** [CMakeFiles/Makefile2:94: CMakeFiles/expat.dir/rule] Error 2
```

Not sure why it's looking for a `.so` file even though we attempted to build a static library? I suppose it could be some AIX weirdness either your cmakelists.txt or CMAKE itself isn't handling?

--- a/expat/CMakeLists.txt
+++ b/expat/CMakeLists.txt
@@ -88,6 +88,15 @@ if(DEFINED BUILD_SHARED_LIBS)
 else()
     set(_EXPAT_SHARED_LIBS_DEFAULT ON)
 endif()
+
+# AIX has none(!) of the high quality entropy sources supported by Expat (e.g.
+# getentropy(3)) but, unlike UNIX, has /dev/urandom availability guaranteed.
+if(AIX)  # needs CMake >=4.0 (or explicit `-DAIX=ON`)
+    set(_EXPAT_DEV_URANDOM_DEFAULT ON)
+else()
+    set(_EXPAT_DEV_URANDOM_DEFAULT OFF)
+endif()
+
 if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE NoConfig)  # so that accessing EXPAT_*_POSTFIX will be waterproof
 endif()
@@ -152,7 +161,7 @@ expat_shy_set(EXPAT_NS ON CACHE BOOL "Define to make XML Namespaces functionalit
 mark_as_advanced(EXPAT_NS)
 expat_shy_set(EXPAT_WARNINGS_AS_ERRORS OFF CACHE BOOL "Treat all compiler warnings as errors")
 if(UNIX OR _EXPAT_HELP)
-    expat_shy_set(EXPAT_DEV_URANDOM OFF CACHE BOOL "Define to include code reading entropy from `/dev/urandom'.")
+    expat_shy_set(EXPAT_DEV_URANDOM ${_EXPAT_DEV_URANDOM_DEFAULT} CACHE BOOL "Define to include code reading entropy from `/dev/urandom'.")
     mark_as_advanced(EXPAT_DEV_URANDOM)
 endif()
 if(UNIX OR WASI OR _EXPAT_HELP)

